apiserver/machinemanager/instance_information.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package machinemanager
+
+import (
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/constraints"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/state"
+	"github.com/juju/juju/state/stateenvirons"
+)
+
+// InstanceTypes returns instance type information for the cloud and region
+// in which the current model is deployed.
+func (mm *MachineManagerAPI) InstanceTypes(cons params.ModelInstanceTypesConstraints) (params.InstanceTypesResults, error) {
+	st := mm.st.(*state.State)
```

Comment:
Please do the same for this method as you did for the other one, and use interfaces (stateInterface).